app/models/sign.rb

```diff
@@ -65,19 +65,18 @@ def self.current_page(per_page, last_result_index, all_result_length)
   end
 
   def self.search(params)
-    before_time = Time.now.to_f
+    before_time = Time.current
     url = url_for_search(params)
     xml_document = Nokogiri::XML(open(url))
     entries = xml_document.css(ELEMENT_NAME)
     count = xml_document.css('totalhits').inner_text.to_i
-    save_time_elapsed(url, before_time, count)
-
+    save_time_elapsed(url, before_time, count) if Rails.application.secrets.track_search_requests?
```

Comment:
Can this go in as a guard clause on `save_time_elapsed`?

Reply:
@joshmcarthur - I am not sure doing a block/yield type method to record time will be tidy, given that the Request record needs to know the count of results as well.
I tried refactoring but I think this is kind of the clearest.